server/pipeline/stepbuilder/stepBuilder.go

```diff
@@ -194,6 +196,27 @@ func (b *StepBuilder) genItemForWorkflow(workflow *model.Workflow, axis matrix.A
 		maps.Copy(item.Labels, b.DefaultLabels)
 	}
 
+	// "woodpecker-ci.org" namespace is reserved for internal use
+	for key := range item.Labels {
+		if strings.HasPrefix(key, pipeline.InternalLabelPrefix) {
+			log.Debug().Str("forge", b.Forge.Name()).Str("repo", b.Repo.FullName).Str("label", key).Msg("dropped pipeline label with reserved prefix woodpecker-ci.org")
+			delete(item.Labels, key)
+		}
+	}
+
```

Comment:
Those labels should not be mixed with the existing labels as those are used for filtering etc. Can we not add a new struct field e.g. `item.InternalLabels` or `item.MetadataLabels` here already?

Reply:
I would say they are also useful for filtering, not the originally intended use, but still useful